specification/policyinsights/resource-manager/Microsoft.PolicyInsights/preview/2018-07-01-preview/policyTrackedResources.json

```diff
@@ -368,6 +368,10 @@
       "enum": [
         "default"
       ],
+      "x-ms-enum": {
+        "name": "PolicyTrackedResourcesResourceType",
+        "modelAsString": false
```

Comment:
```suggestion
        "modelAsString": true
```

Reply:
Are you sure these enums are extensible? Changing this from false to true will bring breaking changes to every SDK right now.

Reply:
From .net sdk code it looks like no matter what this is set to these aren't extensible anyway.  I.e. there is no actual parameter that corresponds with this, its just a hardcoded string in the method... if thats the case then I don't care what modelAsString is set to.  If thats not the case then yes, some of these can absolutely have more values in the future (specifically policyTrackedResourcesResourceParameter, policyEventsResourceParameter, and policyStatesSummaryResource)

Reply:
there is also a consistency issue due to use using "string" enums everywhere else, conversion of enums -> string being slower than string -> string, etc...

Reply:
I am not sure if you are talking about track 1 DotNet SDK - but if you want this value to be a hard-coded value in track 2 DotNet SDK, you have to set this to `false`.
In modeler v4, the modelAsString is not meaning like "if this value is modeled as a string type", instead, this now more means whether this is extensible in the future.
It is controlled by the modeler whether a method will show up in the parameter signature. After a specific version of the modeler, they made the decision that enums with `modelAsString` true will always be in the method signature. Only when modelAsString is false and it has only one value, the modeler will generate the enum parameter to the signature.
If you could confirm here you want the SDK to always hard-code this value, the modelAsString should be false

Reply:
And the new DotNet track 2 SDK introduced implicit operator for the case of `modelAsString: true`: https://github.com/Azure/azure-sdk-for-net/blob/e59d80e92c9409b57f364cc5706f1a144271ccdc/sdk/resources/Azure.ResourceManager.Resources/src/Generated/Models/ApplicationArtifactName.cs#L43
You do not need to explicitly convert string to this type.